docs/DeveloperGuide.md

```diff
@@ -520,6 +607,96 @@ The original AB3 implementation of edit commands, which would default to the ori
+
+![Add Contact Activity Diagram](images/AddContactActivityDiagram.png)
+
+The following shows the sequence diagram of the `AddContactCommand`.
+
+![Add Contact Sequence Diagram](images/AddContactSequenceDiagram.png)
+
+The following steps will describe the execution of the `DeleteContactCommand`, assuming no errors are encountered:
+
+1. When the `DeleteContactCommand` is executed, it will first call the model's `getFilteredContactList` method 
+to determine the last shown list of contacts.
+2. Then, it will call the index's `getZeroBased` method to find the zero-based index of the contact it must delete.
+3. Then, it will check if this index is within range.
+4. If it is, it calls the model's `deleteContact` method.
+5. The model will then call the `removeContact` method of TrackIter, which deletes the contact in question from the app.
+
+![Delete Contact Activity Diagram](images/DeleteContactActivityDiagram.png)
+
+The following shows the sequence diagram of the `DeleteContactCommand`.
+
+![Delete Contact Sequence Diagram](images/DeleteContactSequenceDiagram.png)
+
+The following steps will describe the execution of the `EditContactCommand`, assuming no errors are encountered:
+
+1. When the `EditContactCommand` is executed, it will first call the model's `getFilteredContactList` method 
+to determine the last shown list of contacts.
+2. Then, it will call the index's `getZeroBased` method to find the zero-based index of the contact we must edit.
+5. It will then check if the index is within range.
+6. If it is, it calls the model's `setContact` method.
+7. The model will then call the `setContact` method of TrackIter, which replaces the original contact with the edited
+ version in the app.
+ 
+![Edit Contact Command Activity Diagram](images/EditContactActivityDiagram.png)
+
+The follow shows the sequence diagram of the `EditContactCommand` assuming no errors are encountered.
+
+![Edit Contact Sequence Diagram](images/EditContactSequenceDiagram.png)
+
+The `getModuleContacts` function takes in a Module Code and returns all contacts that have a tag that matches the specified module.
+When `getModuleContacts` is called, it uses the `ContactHasTagPredicate` to update the contact list in the app to only show
+ the contacts that have the desired tag.
+
+This is the sequence diagram of `getModuleContacts`.
+
+![Get Module Contacts Sequence Diagram](images/GetModuleContactsSequenceDiagram.png)
+
+#### Design Considerations
+
+A number of fields in a contact (namely phone number and e-mail address) are optional. In the case they are not specified, 
+ we store them as null. Similar to tasks, we wanted users to be able to remove any optional field simply by 
+  specifying the `/p` or `/e` flag without providing a parameter. Hence, we added 2 additional boolean variables - `isPhoneChanged` and 
+   `isEmailChanged`, to know whether users wanted to remove the existing phone number or e-mail address.
+
```

Comment:
For contacts, maybe you can add an extra diagram and explanation about the getModuleContacts method